File: app/src/main/java/com/customerdynamics/sdk/custom/holders/OutcomingTextAndButtonsMessageViewHolder.java
package com.customerdynamics.sdk.custom.holders;

import android.view.View;
import android.widget.TextView;

import com.customerdynamics.sdk.R;
import com.customerdynamics.sdk.model.Message;
import com.stfalcon.chatkit.messages.MessageHolders;

/*
 * Created by troy379 on 05.04.17.
 */
public class OutcomingTextAndButtonsMessageViewHolder
        extends MessageHolders.OutcomingTextMessageViewHolder<Message> {

    private TextView tvDuration;
    private TextView tvTime;

    public OutcomingTextAndButtonsMessageViewHolder(View itemView, Object payload) {
        super(itemView, payload);
        tvDuration = itemView.findViewById(R.id.duration);
        tvTime = itemView.findViewById(R.id.time);
    }

    @Override
    public void onBind(Message message) {
        super.onBind(message);
        tvDuration.setText("12min.");
        tvTime.setText("custom plugin");
    }
}

File: app/src/main/java/com/nice/cxonechat/sample/custom/holders/OutcomingTextAndButtonsMessageViewHolder.kt
package com.nice.cxonechat.sample.custom.holders

import android.view.View
import android.widget.TextView
import com.nice.cxonechat.sample.R
import com.nice.cxonechat.sample.custom.holders.plugin.bindToViewGroup
import com.nice.cxonechat.sample.model.PluginMessage
import com.stfalcon.chatkit.messages.MessageHolders.OutcomingTextMessageViewHolder
import com.stfalcon.chatkit.utils.DateFormatter
import com.stfalcon.chatkit.utils.DateFormatter.Template.TIME

class OutcomingTextAndButtonsMessageViewHolder(
    itemView: View,
    payload: Any?,
) : OutcomingTextMessageViewHolder<PluginMessage>(itemView, payload) {

    private val tvTime: TextView = itemView.findViewById(R.id.time)

    init {
        bubble = itemView.findViewById(R.id.bubble)
    }

    override fun onBind(message: PluginMessage) {
        super.onBind(message)
        bubble.removeAllViews()
        tvTime.text = DateFormatter.format(message.createdAt, TIME)
        val element = message.content.element
        element.bindToViewGroup(
            parent = bubble,
        ) {
            bindFallback(message)
        }
    }

    private fun bindFallback(message: PluginMessage) = with(bubble) {
        val text = TextView(context)
        text.text = message.content.postback
        addView(text)
    }

}
